src/media/js/apps_buttons.js

```diff
@@ -47,14 +47,27 @@ define('apps_buttons',
 
         // If it's a paid app, ask the user to sign in first.
         if (product.receipt_required && !user.logged_in()) {
+            var payWindow;
+
+            if (!capabilities.navPay) {
+                console.log('Creating payment window ahead of time via user click');
+                payWindow = payments.utils.openWindow({url: ''});
```

Comment:
Just a nit but I think it would be better to call this as ``payments.openWindow()`` and then define it in the exports as ``{..., openWindow: fxpay.utils.openWindow}``.